sdk/schemaregistry/schema-registry-json/samples-dev/schemaRegistryJsonWithValidation.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates the use of JsonSerializer to create messages with json-serialized payload using schema from Schema Registry.
+ */
+
+import { DefaultAzureCredential } from "@azure/identity";
+import {
+  SchemaRegistryClient,
+  SchemaDescription,
+  KnownSchemaFormats,
+} from "@azure/schema-registry";
+import { DeserializeOptions, JsonSerializer } from "@azure/schema-registry-json";
+
+import Ajv from "ajv";
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// The fully qualified namespace for schema registry
+const schemaRegistryFullyQualifiedNamespace =
+  process.env["SCHEMA_REGISTRY_ENDPOINT"] || "<endpoint>";
+
+// The schema group to use for schema registration or lookup
+const groupName = process.env["SCHEMA_REGISTRY_GROUP"] || "AzureSdkSampleGroup";
+
+// Sample Json Schema for user with first and last names
+export const schemaObject = {
+  $schema: "https://json-schema.org/draft/2020-12/schema",
+  $id: "https://example.com/product.schema.json",
+  title: "User",
+  description: "A user for the product",
+  type: "object",
+  properties: {
+    firstName: {
+      type: "string",
+      required: true,
+    },
+    lastName: {
+      type: "string",
+      required: true,
+    },
+  },
+};
+
+// Matching TypeScript interface for schema
+interface User {
+  firstName: string;
+  lastName: string;
+}
+
+const schema = JSON.stringify(schemaObject);
+
+// Description of the schema for registration
+const schemaDescription: SchemaDescription = {
+  name: schemaObject.$id,
+  groupName,
+  format: KnownSchemaFormats.Json,
+  definition: schema,
+};
+
+// Validation using a third party library
+const validateOptions: DeserializeOptions = {
+  validateCallback(message, schema) {
+    const ajv = new Ajv();
+    const validator = ajv.compile(JSON.parse(schema));
```

Comment:
I think we should create those at the time of serialization. How about putting the validate function in a map where the schema's string definition is the key and then access that function here from the map?